docs: add terms page to glossary (#336)
alchemyplatform/aa-sdk: site/glossary/terms.md

---
outline: deep
head:
  - - meta
    - property: og:title
      content: Terms
  - - meta
    - name: description

>       content: Glossary of terms realted to Account Kit

Reviewer: ```suggestion
      content: Glossary of terms related to Account Kit
```

  - - meta
    - property: og:description

>       content: Glossary of terms realted to Account Kit

Reviewer: ```suggestion
      content: Glossary of terms related to Account Kit
```

  - - meta
    - name: twitter:title
      content: Terms
  - - meta
    - name: twitter:description

>       content: Glossary of terms realted to Account Kit

Reviewer: ```suggestion
      content: Glossary of terms related to Account Kit
```

---

# Terms

## Account Kit

> Account Kit is a framework designed to embed smart accounts in web3 applications. It includes a set of tools such as [Signer integrations](https://accountkit.alchemy.com/smart-accounts/signers/choosing-a-signer.html), [Gas Manager](https://docs.alchemy.com/docs/gas-manager-services) and [Bundler](https://docs.alchemy.com/docs/bundler-services) utilities that unlock features such as [gas sponsorship](https://accountkit.alchemy.com/tutorials/sponsoring-gas/sponsoring-gas.html), [batched transactions](https://accountkit.alchemy.com/tutorials/batching-transactions.html) and email/social login. With its user-friendly suite of SDKs, known as [aa-sdk](https://github.com/alchemyplatform/aa-sdk), Account Kit makes it easy to deploy smart accounts, manage user operations, and handle gas sponsorship, streamlining the entire process with minimal coding effort.

Reviewer: ```suggestion
Account Kit is a framework designed to embed smart accounts in web3 applications. It includes a set of tools such as [Signer integrations](https://accountkit.alchemy.com/smart-accounts/signers/choosing-a-signer.html), [Gas Manager](https://docs.alchemy.com/docs/gas-manager-services) and [Bundler](https://docs.alchemy.com/docs/bundler-services) utilities that unlock features such as [gas sponsorship](https://accountkit.alchemy.com/tutorials/sponsoring-gas/sponsoring-gas.html), [batched transactions](https://accountkit.alchemy.com/tutorials/batching-transactions.html) and email/social login. With its user-friendly suite of SDKs, known as [aa-sdk](https://github.com/alchemyplatform/aa-sdk), Account Kit makes it easy to deploy smart accounts, manage `UserOperation`s, and handle gas sponsorship, streamlining the entire process with minimal coding effort.
```

## Bundler

> A network participant that collects and submits user operations to the blockchain, handling the associated gas fees, in exchange for payment during user operation processing either directly from the user or from a [Paymaster](https://www.alchemy.com/overviews/what-is-a-paymaster). Alchemy’s implementation of a bundler is called [Rundler](https://github.com/alchemyplatform/rundler). It is written in Rust and designed to achieve high performance and reliability.

Reviewer: would mention 4337 somewhere here.

Author: Done!

## Entrypoint

> A standardized smart contract that acts as the primary gateway for processing user operations on the blockchain. It receives bundled user operations from Bundlers and verifies and executes these operations according to predefined rules, ensuring security and adherence to user-specified conditions.

Reviewer: ```suggestion
A standardized smart contract that acts as the primary gateway for processing `UserOperation`s on the blockchain. It receives bundled user operations from Bundlers and verifies and executes these operations according to predefined rules, ensuring security and adherence to user-specified conditions.
```

## Plugin

A module for [ERC-6900](https://eips.ethereum.org/EIPS/eip-6900) smart accounts, enabling specific functions like validation, execution, and hooks. These plugins ensure modularity, upgradeability, and adherence to standardized interfaces.

## Provider

An intermediary or connector that enables interactions between applications and a blockchain network. Providers offer the necessary infrastructure and APIs to connect, query, and interact with the blockchain, enabling transactions, smart contract executions, and data retrieval. You can use `AlchemyProvider` within [aa-sdk](https://github.com/alchemyplatform/aa-sdk) to query blockchain data and send user operations.

## Signer

> A service or application that manages the private key and signs user operations. Types of signers include:

Reviewer: ```suggestion
A service or application that manages the private key and signs `UserOperation`s. Types of signers include:
```

- **Custodial**: Managed by a third party, it holds and autonomously uses the private key for transactions, necessitating complete user trust.
- **Non-custodial**: While a third party manages the private key, user involvement is required for signing transactions. Examples: Turnkey, Magic.
- **MPC (Multi-Party Computation)**: Partial or complete key shares are managed by a third party, but user participation is needed for transaction signatures. Examples: Privy, Portal, Fireblocks NCW.
- **Decentralized MPC**: Operated by a decentralized network, it manages key shares and requires node consensus for transaction signatures. Examples: Lit, Web3auth, 0xpass.

## Smart Account

> A [smart account](https://accountkit.alchemy.com/smart-accounts/accounts/choosing-a-smart-account.html#what-s-a-smart-account) is an individual onchain account located at a public address where an ERC-4337 smart contract account is deployed. This address is controlled by one or more owners of the smart contract account. The [aa-sdk](https://github.com/alchemyplatform/aa-sdk) supports different smart account implementations such as [Light Account](https://accountkit.alchemy.com/smart-accounts/accounts/guides/light-account.html), [Simple Account](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/samples/SimpleAccount.sol) and [Modular Account](https://accountkit.alchemy.com/smart-accounts/accounts/guides/modular-account.html) (coming soon). You can also [add add your own account implementation in aa-sdk](https://accountkit.alchemy.com/smart-accounts/accounts/contributing.html).

Reviewer: ```suggestion
A [smart account](https://accountkit.alchemy.com/smart-accounts/accounts/choosing-a-smart-account.html#what-s-a-smart-account) is an individual on-chain account located at a public address where an ERC-4337 smart contract account is deployed. This address is controlled by one or more owners of the smart contract account. The [aa-sdk](https://github.com/alchemyplatform/aa-sdk) supports different smart account implementations such as [Light Account](https://accountkit.alchemy.com/smart-accounts/accounts/guides/light-account.html), [Simple Account](https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/samples/SimpleAccount.sol) and [Modular Account](https://accountkit.alchemy.com/smart-accounts/accounts/guides/modular-account.html) (coming soon). You can also [add add your own account implementation in aa-sdk](https://accountkit.alchemy.com/smart-accounts/accounts/contributing.html).
```

> ## User Operation

Reviewer: ```suggestion
## `UserOperation`
```

A pseudo-transaction object introduced by the ERC-4337 standard, used to execute actions via a smart account. It encapsulates the intended actions or transactions of the user, which are executed on-chain by a [Bundler](https://docs.alchemy.com/docs/bundler-services).

## Wallet

A software application to manage one or more accounts. It supports connecting to web3 apps and signing transactions via the [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) interface.

## Wallet-as-a-Service (WaaS)

> Also called a key management service (KMS). WaaS is a software as a service provider that stores private key material. A WaaS provider will be classified under one of the [signer custody types](#signer).

Reviewer: ```suggestion
Also called a key management service (KMS). WaaS is a software as a service provider that stores private key material. A WaaS provider would be classified under one of the [signer custody types](#signer).
```
